docs/USER_RECORD.md

```diff
@@ -554,6 +554,15 @@ it is bypassed.
 auto-login. Systems are supposed to automatically log in a user marked this way
 during boot, if there's exactly one user on it defined this way.
 
+`sessionType` → A string that indicates the user's preferred session type (i.e.
+`x11`, `wayland`, or other values valid for `$XDG_SESSION_TYPE`). This should be
+used by the display manager to pre-select the correct environment to log into.
+
+`sessionName` → A string that indicates the user's preferred session (i.e. `gnome`,
+`gnome-classic`, `plasma`, `kodi`, or others that appear in locations such as
+`/usr/share/xsessions` or `/usr/share/wayland-sessions`). This should be used by
+the display manager to pre-select the correct environment to log into.
```

Comment:
sessionName is a bit of a confusing name for this. we give session identifiers after all, which sounds like this is what it is.

moreover we actually have a concept for this already in logind's session "desktop" field, and the XDG_SESSION_DESKTOP env var.

Also, these are named as if they indicate a fact, but they just indicate a preference, no? the user can pick whatever they want in theory at login time. This stuff is just about enforcing a preference that if the user doesn't do that.

Maybe names such as preferredSessionType and preferredSessionDesktop would be better?

Reply:
alternatively: defaultSessionType, defaultSessionDesktop?